### PR TITLE
Added messages to run & complete pings

### PR DESCRIPTION
**Pull request by trammel**
The run & complete pings can also take message arguments; this commit
adds tests around that, and then implements the change.

Additional minor styling changes for hashes for compatability.

### CODE REVIEW

File: lib/cronitor.rb [hunk 1]
```diff
@@ -34,7 +34,7 @@ def initialize(token: nil, opts: {}, code: nil)
   def create
     response = Unirest.post(
       "#{API_URL}/monitors",
-      headers: default_headers.merge 'Content-Type' => 'application/json',
+      headers: default_headers.merge('Content-Type': 'application/json'),
```
Comment (eherot) [inline lib/cronitor.rb]: This will make `'Content-Type'` into a symbol. I don't think you want that. E.g.:
```ruby
[4] pry(main)> default_headers = {}
=> {}
[5] pry(main)> default_headers.merge('Content-Type': 'application/json')
=> {:"Content-Type"=>"application/json"}
```
Also, this is just a stylistic change and probably shouldn't be included in this pull request.

File: lib/cronitor.rb [hunk 2]
```diff
@@ -105,6 +105,6 @@ def server_error?(response)
   end
 
   def default_headers
-    { 'Accept' => 'application/json' }
+    { 'Accept': 'application/json' }
```
Comment (eherot) [inline lib/cronitor.rb]: See above comment...

File: lib/cronitor.rb [hunk 3]
```diff
@@ -57,7 +57,7 @@ def exists?(name)
 
   def ping(type, msg = nil)
     url = "#{PING_URL}/#{code}/#{type}"
-    url += "?msg=#{URI.escape msg}" if type == 'fail' && !msg.nil?
+    url += "?msg=#{URI.escape msg}" if ['run', 'complete', 'fail'].include?(type) && !msg.nil?
```
Comment (eherot) [inline lib/cronitor.rb]: Might as well just remove `['run', 'complete', 'fail'].include?(type)` since that's all the available types anyway

File: spec/support/fake_cronitor.rb [hunk 4]
```diff
@@ -12,18 +16,24 @@ class FakeCronitor < Sinatra::Base
   post '/v1/monitors' do
     payload = JSON.parse request.body.read
     # Check that we have the necessary payload values very simply
-    %w(name rules notifications).each do |k|
+    %w[name rules notifications].each do |k|
```
Comment (eherot) [inline spec/support/fake_cronitor.rb]: more stylistic changes

File: spec/support/fake_cronitor.rb [hunk 5]
```diff
@@ -12,18 +16,24 @@ class FakeCronitor < Sinatra::Base
...
       return json_response 400, "invalid_no_#{k}" unless payload.key? k
     end
     json_response 200, 'new_monitor'
   end
 
-  %w(run complete fail).each do |ping|
+  %w[run complete fail].each do |ping|
```
Comment (eherot) [inline spec/support/fake_cronitor.rb]: more stylistic changes

File: spec/support/fake_cronitor.rb [hunk 6]
```diff
@@ -12,18 +16,24 @@ class FakeCronitor < Sinatra::Base
...
+      content_type :json
+      status 200
+      params.to_json
     end
   end
 
+  before do
+    self.class.requests = [self.class.requests, request]
+  end
+
```
Comment (eherot) [inline spec/support/fake_cronitor.rb]: What does this do?

**Comment by eherot** [comment]
Hi @trammel, thank you very much for the PR! I'm not sure it really makes sense to include most of these style changes with a PR like this and also some of them seem to do extra things we maybe don't want. Would you be willing to revise this to leave those out? Maybe we can address them separately if you really feel they should be included...

File: lib/cronitor.rb (hunk 1 above)
Comment (trammel) [inline lib/cronitor.rb]: The additional brackets seem to be important, as some versions of ruby were refusing to compile the code without them. At least on my machine. The string to symbol change was unintentional.

File: lib/cronitor.rb (hunk 3 above)
Comment (trammel) [inline lib/cronitor.rb]: 'run', 'complete' & 'fail' all accept messages. 'pause' doesn't according to these docs https://cronitor.io/docs/understanding-ping-urls

File: lib/cronitor.rb (hunk 2 above)
Comment (trammel) [inline lib/cronitor.rb]: Will revert.

File: spec/support/fake_cronitor.rb (hunk 4 above)
Comment (trammel) [inline spec/support/fake_cronitor.rb]: I ran rubocop over the file to find problems, and accidentally auto-fixed them. Will revert.

File: spec/support/fake_cronitor.rb (hunk 6 above)
Comment (trammel) [inline spec/support/fake_cronitor.rb]: The changes to /get was a mistake. Will revert.

The before method was introduced to allow the capturing or requests, and then be able to assert that the requests did in fact pass the message on to the called API in tests.

File: lib/cronitor.rb (hunk 3 above)
Comment (eherot) [inline lib/cronitor.rb]: ohhhh good point. We should still probably do shorthand format though:
```ruby
%w(run complete fail)
```

File: spec/support/fake_cronitor.rb (hunk 6 above)
Comment (eherot) [inline spec/support/fake_cronitor.rb]: Ah ok. Would you mind adding a code comment to that effect? The code itself is a little...opaque.
